edb/schema/scalars.py

```diff
@@ -309,8 +309,9 @@ def as_alter_delta(
 
         # If this is an enum and enum_values changed, we need to
         # generate a rebase.
+        old_enum_values = self.get_enum_values(self_schema)
         enum_values = alter.get_local_attribute_value('enum_values')
-        if enum_values is not None:
+        if old_enum_values and enum_values:
```

Comment:
Actually, what if we removed the `enum_values` attribute set from `alter` in this case?
That might eliminate the cursedness by fully pushing things down to the rebase, even in the delta case.